### PR TITLE
Improve efficiency in FindRenderizableReferenceNumeric and getReferences

### PR DESCRIPTION
**Pull request by zeripath**

### CODE REVIEW

File: modules/references/references.go [hunk 1]
```diff
@@ -14,6 +15,7 @@ import (
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/markup/mdstripper"
 	"code.gitea.io/gitea/modules/setting"
+	"github.com/yuin/goldmark/util"
```
Comment (techknowlogick) [inline modules/references/references.go:19]:
```suggestion

 	"github.com/yuin/goldmark/util"
```

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
@techknowlogick fancy re-review?

**Review by techknowlogick — APPROVED** [review]

File: modules/references/references.go [hunk 2]
```diff
@@ -465,18 +468,17 @@ func findAllIssueReferencesBytes(content []byte, links []string) []*rawReference
...
+	isPull := content[start+sep] == '!'
 	if prOnly && !isPull {
 		return nil
 	}
-	repo := refid[:sep]
-	issue := refid[sep+1:]
-	index, err := strconv.ParseInt(issue, 10, 64)
+	repo := string(content[start : start+sep])
+	issue := string(content[start+sep+1 : end])
+	index, err := strconv.ParseInt(string(issue), 10, 64)
```
Comment (KN4CK3R) [inline modules/references/references.go:481]: `issue` is a string

Comment (zeripath) [reply modules/references/references.go:481]: Yup. Oh well. It should just get compiled away. Next time I'm in the area I'll remove that superfluous cast.

Comment (6543) [reply modules/references/references.go:481]: yes saw your comment after merge :/

**Pull request merged**